app/hooks/query.ts

```diff
@@ -82,6 +83,11 @@ export const useQueryArtistTopSongs = (artistName?: string) => {
   return querySuccess ? query : backupQuery
 }
 
+export const useQueryArtistAllSongs = (id: string) => {
+  const fetchArtistAllSongs = useFetchArtistAllSongs()
+  return useQuery(qk.artistAllSongs(id), () => fetchArtistAllSongs(id))
```

Comment:
Removing the placeholder data in here was the right call, I don't think it needs it.